viperx/vars.go

```diff
@@ -20,8 +20,24 @@ func UnmarshalKey(key string, destination interface{}) error {
 	}
 
 	var b bytes.Buffer
-	if err := json.NewEncoder(&b).Encode(mapx.ToJSONMap(viper.Get(key))); err != nil {
+
+	// This may be a string in the case where a value was provided via an env var.
+	// If it's a string, try to decode it as json.
+	if v, ok := value.(string); ok {
+		if _, err := b.WriteString(v); err != nil {
+			return errors.WithStack(err)
+		}
+
+		// Try decoding the json directly
+		if err := json.NewDecoder(&b).Decode(destination); err == nil {
```

Comment:
You could use `github.com/tidwall/gjson.Valid` for a faster and less memory expensive operation here!

Reply:
Valid would not decode it into `destination` though :thinking:

Reply:
Ah yeah, I misread the error handling - though it is `err != nil`

Reply:
I'll make the comment a bit clearer.